src/components/composer/composer/composer.ts

```diff
@@ -437,6 +437,7 @@ export class Composer extends Component<ComposerProps, SpreadsheetChildEnv> {
     if (this.env.model.getters.isReadonly()) {
       return;
     }
+    this.composerRef.el?.focus();
```

Comment:
This line is added to fix the bug: when the context menu/topbar menu is opened, clicking the topbar composer will throw error. But it's not perfect: the first click on the topbar composer won't set the cursor position accurately. I tried to fix it completely but failed; however I don't think it's a big problem so I keep it.